opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java

```diff
@@ -154,7 +154,7 @@ public static void createAdditionalColumns(SQLiteDatabase db) {
     public static void addEventLocationId(SQLiteDatabase db) {
         DatabaseMigrationUtils.addColumnIfNotExists(db, Table.event.name(), client_column.locationId.name(), VARCHAR);
         DatabaseMigrationUtils.addIndexIfNotExists(db, Table.event.name(), client_column.locationId.name());
-
+        db.execSQL(String.format("UPDATE %s set %s = %s", Table.event.name(), event_column.locationId.name(), "substr(json,instr(json, '\"locationId\":')+14,36)"));
```

Comment:
This works perfect except for situations where the event does not have a `locationId`. I don't know if that is possible on Reveal currently

Reply:
Added a filter for only events with locationId